projects/packages/forms/src/contact-form/class-contact-form-field.php

```diff
@@ -301,33 +333,94 @@ public function render() {
-		}
-		if ( is_numeric( $this->get_attribute( 'borderwidth' ) ) ) {
-			$this->block_styles .= '--jetpack--contact-form--border-size: ' . esc_attr( $this->get_attribute( 'borderwidth' ) ) . 'px;';
-			$this->field_styles .= 'border-width: ' . (int) $this->get_attribute( 'borderwidth' ) . 'px;';
-		}
-		if ( is_numeric( $this->get_attribute( 'lineheight' ) ) ) {
-			$this->block_styles  .= '--jetpack--contact-form--line-height: ' . esc_attr( $this->get_attribute( 'lineheight' ) ) . ';';
-			$this->field_styles  .= 'line-height: ' . (int) $this->get_attribute( 'lineheight' ) . ';';
-			$this->option_styles .= 'line-height: ' . (int) $this->get_attribute( 'lineheight' ) . ';';
-		}
-		if ( ! empty( $this->get_attribute( 'bordercolor' ) ) ) {
-			$this->block_styles .= '--jetpack--contact-form--border-color: ' . esc_attr( $this->get_attribute( 'bordercolor' ) ) . ';';
-			$this->field_styles .= 'border-color: ' . esc_attr( $this->get_attribute( 'bordercolor' ) ) . ';';
-		}
-		if ( ! empty( $this->get_attribute( 'inputcolor' ) ) ) {
-			$this->block_styles  .= '--jetpack--contact-form--text-color: ' . esc_attr( $this->get_attribute( 'inputcolor' ) ) . ';';
-			$this->block_styles  .= '--jetpack--contact-form--button-outline--text-color: ' . esc_attr( $this->get_attribute( 'inputcolor' ) ) . ';';
-			$this->field_styles  .= 'color: ' . esc_attr( $this->get_attribute( 'inputcolor' ) ) . ';';
-			$this->option_styles .= 'color: ' . esc_attr( $this->get_attribute( 'inputcolor' ) ) . ';';
-		}
-		if ( ! empty( $this->get_attribute( 'fieldbackgroundcolor' ) ) ) {
-			$this->block_styles .= '--jetpack--contact-form--input-background: ' . esc_attr( $this->get_attribute( 'fieldbackgroundcolor' ) ) . ';';
-			$this->field_styles .= 'background-color: ' . esc_attr( $this->get_attribute( 'fieldbackgroundcolor' ) ) . ';';
+		$label_classes  = $this->get_attribute( 'labelclasses' );
+		$label_styles   = $this->get_attribute( 'labelstyles' );
+		$input_classes  = $this->get_attribute( 'inputclasses' );
+		$input_styles   = $this->get_attribute( 'inputstyles' );
+		$option_classes = $this->get_attribute( 'optionclasses' );
+		$option_styles  = $this->get_attribute( 'optionstyles' );
+
+		$has_block_support_styles = ! empty( $label_classes ) || ! empty( $label_styles ) || ! empty( $input_classes ) || ! empty( $input_styles ) || ! empty( $option_classes ) || ! empty( $option_styles );
+
+		if ( $has_block_support_styles ) {
+			// Do any of the block support classes need to be applied at the field wrapper level? Do we need to make the classes etc filterable as per the field classes?
+
+			// Classes.
+			if ( ! empty( $label_classes ) ) {
+				$this->label_classes .= esc_attr( $label_classes );
+			}
+			if ( ! empty( $input_classes ) ) {
+				$class              .= $class ? ' ' . esc_attr( $input_classes ) : esc_attr( $input_classes );
+				$this->field_classes = $input_classes;
+			}
+			if ( ! empty( $option_classes ) ) {
```

Comment:
Non urgent question:

In my testing I'm seeing that `optionclasses` passed to `Contact_Form_Field` will be added to the field wrapper element. 

```php
// Attributes passed to `new Contact_Form_Field()`
$attributes = array(
    ....some other attrs
    'optionclasses' => 'has-cheese',
);
```

```html
<div style="" class='grunion-field-checkbox-multiple-wrap lalala-wrap has-cheese-wrap grunion-field-wrap'  >
<fieldset ...>
```

Is that intentional?

Reply:
Actually forget it. Sorry, I didn't see the `-wrap` suffix.